src/SubtitleOctopus.cpp

```diff
@@ -19,50 +19,59 @@
 
 int log_level = 3;
 
-typedef struct {
-    void *buffer;
-    int size;
-    int lessen_counter;
-} buffer_t;
+class ReusableBuffer {
+public:
+    ReusableBuffer(): buffer(NULL), size(-1), lessen_counter(0) {}
 
-void* buffer_resize(buffer_t *buf, int new_size, int keep_content) {
-    if (buf->size >= new_size) {
-        if (buf->size >= 1.3 * new_size) {
-            // big reduction request
-            buf->lessen_counter++;
-        } else {
-            buf->lessen_counter = 0;
-        }
-        if (buf->lessen_counter < 10) {
-            // not reducing the buffer yet
-            return buf->buffer;
-        }
+    ~ReusableBuffer() {
+        free(buffer);
     }
 
-    void *newbuf;
-    if (keep_content) {
-        newbuf = realloc(buf->buffer, new_size);
-    } else {
-        newbuf = malloc(new_size);
+    void clear() {
+        free(buffer);
+        buffer = NULL;
+        size = -1;
+        lessen_counter = 0;
     }
-    if (!newbuf) return NULL;
 
-    if (!keep_content) free(buf->buffer);
-    buf->buffer = newbuf;
-    buf->size = new_size;
-    buf->lessen_counter = 0;
-    return buf->buffer;
-}
+    void *take(int new_size, bool keep_content) {
```

Comment:
I tend to `get` for some reason :man_shrugging: 
This is more like `reserve`, because the buffer size will not be smaller than `new_size`.

Reply:
How about `request_resize`, `ensure_size` or `reserve_size`?
`get` is imho to confusing as many array and list types use `get`-function to access individual members.

Reply:
Of these 3 options, `ensure_size` is probably the best fit.

I also thought about `allocate`, but this may confuse about the need of `free`.

We can stick to the original name `resize` :man_shrugging: 

Furthermore, I am thinking of adding a `size` method to return the passed/calculated `new_size` so that it is not recalculated later (see #124).

Reply:
Let’s use `ensure_size` then.